Dependencies/CMakeExtensions/CMakeLists.txt

```
@@ -123,4 +123,20 @@ function(set_win32_arch)
     else()
         message(FATAL_ERROR "Unrecognized compiler: ${CMAKE_CXX_COMPILER}")
     endif()
+endfunction()
+
+# Uses the nuget.config and packages.config files in the current directory to download packages from NuGet.
+# NUGET_PATH will be set to the directory where the packages are installed.
+function (nuget_download)
```

nit: most functions put the verb first. maybe this?
```suggestion
function (download_nuget)
```